src/Contract/Crypto/Schnorr.purs

```diff
@@ -0,0 +1,54 @@
+-- | A module that implements crypto primitives that match CIP-49 SECP256k1
+-- | Schnorr spec.
+module Contract.Crypto.Schnorr
+  ( module X
+  , verifySchnorrSecp256k1Signature
+  , signSchnorrSecp256k1
+  , deriveSchnorrSecp256k1PublicKey
+  , mkSchnorrPublicKey
+  , unSchnorrPublicKey
+  ) where
+
+import Prelude
+
+import Ctl.Internal.Types.ByteArray (ByteArray)
+import Data.Maybe (Maybe)
+import Data.Newtype (unwrap, wrap)
+import Effect.Aff (Aff)
+import Noble.Secp256k1.ECDSA (PrivateKey)
+import Noble.Secp256k1.Schnorr (PrivateKey, SchnorrPublicKey, SchnorrSignature) as X
+import Noble.Secp256k1.Schnorr
+  ( SchnorrPublicKey
+  , SchnorrSignature
+  , getSchnorrPublicKey
+  , signSchnorr
+  , verifySchnorr
+  )
+import Noble.Secp256k1.Schnorr (mkSchnorrPublicKey, unSchnorrPublicKey) as ECDSA
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | Verify arbitrary binary messages signed using the Schnorr signature scheme
+-- | on the SECP256k1 curve.
+-- | Matches CIP-49 spec:
+-- | https://github.com/cardano-foundation/CIPs/blob/master/CIP-0049/README.md
+verifySchnorrSecp256k1Signature
+  :: SchnorrPublicKey -> ByteArray -> SchnorrSignature -> Aff Boolean
+verifySchnorrSecp256k1Signature publicKey message signature =
+  verifySchnorr signature (unsafeCoerce message) publicKey
```

Comment:
When unsafeCoercing, I prefer to make a typed helper function in a where clause, so that if types are changed, the unsafe coersion requires re-evaluating. Eg:

```purescript
verifySchnorrSecp256k1Signature publicKey message signature =
  verifySchnorr signature (coerceMessage message) publicKey
  where
  coerceMessage :: ByteArray -> ???
  coerceMessage = unsafeCoerce
```